Replaced older components in ResourceUpdateDetails page with Shadcn component

NikhilA8606 opened the pull request:
## Replaced four older components with shadcn

- Fixes #10048
- Replaced RadioFormField with RadioGroup
- Replaced TextAreaFormField with Textarea
- Replaced TextFormField with Input
- Replaced FacilitySelect with Autocomplete
- Replaced SelectFormField with Select

@rithviknishad 

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

manmeetnagii committed:
Fixed alignment of components in mobile view (#10038)

dependabot[bot] committed:
Bump @radix-ui/react-toast from 1.2.4 to 1.2.5 (#10116)

dependabot[bot] committed:
Bump @sentry/browser from 8.50.0 to 8.51.0 (#10117)

tellmeY18 committed:
switched to ubuntu-24.04-arm runners (#10026)

Mahendar0701 committed:
Fix : Date field overflow in exception form in mobile view (#10112)

dependabot[bot] committed:
Bump @rollup/rollup-linux-arm64-gnu from 4.30.1 to 4.31.0 (#10065)

Bumps [@rollup/rollup-linux-arm64-gnu](https://github.com/rollup/rollup) from 4.30.1 to 4.31.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.30.1...v4.31.0)

---
updated-dependencies:
- dependency-name: "@rollup/rollup-linux-arm64-gnu"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Rishith25 committed:
Fix: Incorrect Icon in Patient nav bar | Facility switcher hovering and some minor fixes (#10080)

Rishith25 committed:
Added Skeleton Loader for Care public page | Adjusted the roundness of Avatar (#10124)

rajku-dev committed:
Fix: Unwanted scrolling and spacing on availability page (#10130)

NikhilA8606 committed:
Added `date-field` reusable component; Migrated old DateFormFields to new UI components (#10049)

rithviknishad committed:
fixes dob formatting of patient registration (#10135)

rithviknishad committed:
fix unintended error message in questionnaire (#10151)

dependabot[bot] committed:
Bump @rollup/rollup-linux-x64-gnu from 4.30.1 to 4.32.0 (#10141)

Bumps [@rollup/rollup-linux-x64-gnu](https://github.com/rollup/rollup) from 4.30.1 to 4.32.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.30.1...v4.32.0)

---
updated-dependencies:
- dependency-name: "@rollup/rollup-linux-x64-gnu"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Add renovate.json (#10132)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rajku-dev committed:
Change `organizations_remove` route method to DELETE (#10159)

renovate[bot] committed:
Update dependency vite to v5.4.12 [SECURITY] (#10152)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @tailwindcss/forms to v0.5.10 (#10161)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

AdityaJ2305 committed:
Upgrade to `Eslint` v9 (#10102)

renovate[bot] committed:
Update dependency @tailwindcss/forms to v0.5.10 (#10168)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency vite to v5.4.12 [SECURITY] (#10167)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @types/node to v22.10.10 (#10163)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @tailwindcss/typography to v0.5.16 (#10162)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

sainak committed:
Fix dev server (#10174)

Jacobjeevan committed:
Added required field validation and scroll before submit for questionnaire forms (#10169)

Jacobjeevan committed:
reset questionnaire errors (#10177)

rithviknishad committed:
Update renovate config (#10184)

renovate[bot] committed:
Update all dependencies (minor, patch) (#10185)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

NikhilA8606 committed:
Replaced old components in Resource Details Update page to shadcn component

NikhilA8606 committed:
Replace older components in ResourceDetailsUpdate page with shadcn components

gigincg committed:
Prefill Structured Data; Add Edit Links for Structured (#9992)

Co-authored-by: Amjith Titus <amjithtitus09@gmail.com>

rithviknishad committed:
Trigger default filters apply logic only when there are no search params already applied (#10204)

rithviknishad committed:
fixes year_of_birth post patient registration success (#10206)

nihal467 committed:
New Cypress Test for Creation Of Questionnaire (#10210)

gigincg committed:
Remove Query Invalidation after Questionnaire Batch Request

rithviknishad commented on src/components/Resource/ResourceDetailsUpdate.tsx:
```diff
@@ -64,6 +71,31 @@ export const ResourceDetailsUpdate = (props: resourceProps) => {
+    };
+
+    retriveData();
+  }, [data]);
```
Let's not use `request`+`useEffect`+`useState`. Let's use `useQuery` instead to achieve data fetching.

Refer: https://docs.ohc.network/docs/care/development/data-fetching-in-care/

rithviknishad commented on src/components/Resource/ResourceDetailsUpdate.tsx:
```diff
@@ -200,22 +226,34 @@ export const ResourceDetailsUpdate = (props: resourceProps) => {
+                }
+              >
+                <SelectTrigger className="mt-2">
+                  <span>{state.form.status || "Select an option"}</span>
```
missing i18n translations btw

rithviknishad commented on src/components/ui/autocomplete.tsx:
```diff
@@ -18,7 +18,7 @@ import {
 } from "@/components/ui/popover";
 
-interface AutoCompleteOption {
+export interface AutoCompleteOption {
   label: string;
   value: string;
 }
```
this change won't be needed once the previously mentioned thing is done

khavinshankar committed:
changed latitude longitude type to a number in facility create form (#10147)

khavinshankar committed:
added i18n support for plugins (#10182)

Jacobjeevan committed:
Confirm Alert for Patient Registration (#10205)

rajku-dev committed:
Render multi line text in Questionnaire Response (#10216)

dependabot[bot] committed:
Bump lucide-react from 0.473.0 to 0.474.0 (#10209)

Bumps [lucide-react](https://github.com/lucide-icons/lucide/tree/HEAD/packages/lucide-react) from 0.473.0 to 0.474.0.
- [Release notes](https://github.com/lucide-icons/lucide/releases)
- [Commits](https://github.com/lucide-icons/lucide/commits/0.474.0/packages/lucide-react)

---
updated-dependencies:
- dependency-name: lucide-react
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update all dependencies (minor, patch) (#10222)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rithviknishad committed:
disable dependabot updates in favour of renovate (#10223)

renovate[bot] committed:
Update actions/github-script action to v7 (#10193)



Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Aakash Singh <mail@singhaakash.dev>

renovate[bot] committed:
Update actions/checkout action to v4 (#10189)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

khavinshankar committed:
fixed inputs focus issue in plugs when a dialog is child of dropdown menu (#10227)

renovate[bot] committed:
Update dependency marked to v15 (#10225)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

modamaan committed:
Fix: Adjust Image Upload Button Width in Edit Cover Photo and Avatar Modal for Mobile View (#10198)

renovate[bot] committed:
Update actions/stale action to v9 (#10194)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

shivankacker committed:
added dpg badge to readme (#10134)

rithviknishad committed:
Switch to `PhoneInput` (#10214)

rithviknishad committed:
Remove 'discharge' from MedicationAdministration category types (#10232)

rithviknishad committed:
Enhance Practitioner Selection in Book Appointment Page (#10236)

rithviknishad committed:
Remove slot type selection from schedule template forms (#10238)

Fixes #10235

rithviknishad committed:
Refactor Token Slot Button into Reusable Component (#10244)

Extract common slot button rendering logic into a shared TokenSlotButton component across Appointments and Public Appointments pages

rithviknishad committed:
prevent showing callout when slots are negative (#10242)

AdityaJ2305 committed:
Follow Up: Upgrade `Eslint` version to v9 (#10180)

renovate[bot] committed:
Update dependency eslint-config-prettier to v10 (#10197)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rithviknishad committed:
Fixes incorrect count in appointments column when search query is present (#10247)

rithviknishad committed:
Remove status change dropdown from AppointmentActions component (#10248)

renovate[bot] committed:
Update all dependencies (minor, patch) (#10245)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

AdityaJ2305 committed:
Disable Unnecessary Browser Pop-up on Patient Registration Form (#10251)

Jacobjeevan committed:
Patient Age and Facility Location Edits (#10260)

modamaan committed:
Fix Layout Overflow on Appointment and Patient Details Pages for Mobile Screens (#10267)

Rishith25 committed:
Fix: Facility Creation Form: Pincode Autofill Overwrites Pre-filled Geographic Data & Added Similar Approach in User Creation Form (#10166)

renovate[bot] committed:
Update dependency prettier-plugin-tailwindcss to v0.6.11 (#10179)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update all dependencies (minor, patch) (#10268)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency vite-plugin-pwa to ^0.21.0 (#10188)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update docker/setup-buildx-action action to v3 (#10271)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update docker/build-push-action action to v6 (#10270)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update leonsteinhaeuser/project-beta-automations action to v2 (#10274)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

AdityaJ2305 committed:
Feat: Display total no. of entities with heading in Organization (#10092)

Jacobjeevan committed:
Patient Detail Tabs: Org List Access (#10121)

NikhilA8606 committed:
Replaced all TextAreaForm component with Textarea shadcn ui component (#10277)

renovate[bot] committed:
Update prince-chrismc/label-merge-conflicts-action action to v3 (#10276)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency sonner to v1.7.3 (#10281)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

bodhish committed:
Add location pages

AdityaJ2305 committed:
Add Real-Time Validation for `CreateUserForm` and Standardize UI Across Forms (#10054)

DonXavierdev committed:
Fix: Adjust spacing and overflow behavior in FileUploadDialog component (#10133)

modamaan committed:
Fix the accessibility issue by adding DialogTitle to DialogContent and making the facility title responsive (#10106)

renovate[bot] committed:
Update dependency ubuntu to v24 (#10229)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

bodhish committed:
Add facility settings routes and layout

gigincg committed:
Fix type edit in QuestionnaireEditor

gigincg committed:
Nested Routes for Facility Settings; Org to Departments

bodhish committed:
Update dependency undici to v5.28.5

nihal467 committed:
patient registration in multiple combination (#10287)

gigincg committed:
Reorganize Facility Settings

bodhish committed:
Simplify GeneralSettings to use FacilityHome component

rithviknishad committed:
Update theme color; update icons; remove unused assets (#10292)

bodhish committed:
Add overview page

gigincg committed:
Refactor Facility Settings; Location Organizations

nihal467 committed:
added * to all mandatory fields in resource create (#10314)

nihal467 committed:
Expand the cypress to search a patient and create an encounter (#10317)

AdityaJ2305 committed:
Fix: View Facility Redirection in public page is broken (#10318)

renovate[bot] committed:
Update all dependencies (minor, patch) (#10290)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Prakhar29Sharma committed:
showing "-" in the emergency contact person field to represent the field is null (#10323)

nihal467 committed:
fixed the redirection in organisation (#10328)

rithviknishad committed:
Fix: Use headers.set() instead of headers.append() in makeHeaders (#10333)

rithviknishad committed:
Set theme color to white in vite PWA manifest (#10321)

rithviknishad committed:
Expose dev server to host (#10330)

rithviknishad committed:
Remove automatic picker show for date and time inputs (#10331)

gigincg committed:
fix Page in location: infer location.mode;

Jacobjeevan committed:
Rm whatsapp phone number, hide email for updates (#10334)

Jacobjeevan committed:
Dob/Age and Lat/Log Override (#10280)

rithviknishad committed:
Remove pin-code autofill feature (#10291)

rithviknishad committed:
fix responsiveness of date field and gender in public patient registration form (#10294)

rithviknishad committed:
Make autocompletes mobile friendly (#10336)

bodhish committed:
Add empty parent parameter to location and organization list queries

nihal467 committed:
removed hard corded values in the appointments (#10337)

bodhish committed:
Refactor location list views with reusable LocationCard component

ashwinpraveengo committed:
fix: Align and format cards in Health Profile (#10298)

rajku-dev committed:
Fix Camera Capture and Upload Workflow (#10326)

AdityaJ2305 committed:
Fix: Resource Letter printing issue  (#10230)

* fix resource letter print crash

* add TFunction

* Empty-Commit

* created new print comp for resource letter

* replace classNames with cn

* rm data from top

* Empty-Commit

* Empty-Commit

* Empty-Commit

rithviknishad committed:
Remove index-based numbering from appointments table row (#10338)

Fixes #10307

amjithtitus09 committed:
Patient Updates Tab, Structured responses filter by Encounter (#10329)

amjithtitus09 committed:
Location Update fix (#10339)

Jacobjeevan committed:
Resources redesign (#10308)

Co-authored-by: rithviknishad <mail@rithviknishad.dev>

Jacobjeevan committed:
ResourceUpdates Bug Fix (#10341)

renovate[bot] committed:
Update dependency jsdom to v26 (#10224)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency tailwind-merge to v3 (#10300)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update all dependencies (minor, patch) (#10343)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rithviknishad committed:
Cleanup: Remove unnecessary GitHub workflows and update linter Node.js version (#10342)

NikhilA8606 committed:
Replaced older components in the useFileManager with shadcn component (#10344)

renovate[bot] committed:
Update all dependencies (minor, patch) (#10346)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mdimado committed:
Replace diary_free with dairy_free (#10187)

renovate[bot] committed:
Update github/ossar-action to v1 (#10153)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Bodhish Thomas <bodhish@gmail.com>

rithviknishad committed:
fix missing wave image files (#10357)

AdityaJ2305 commented:
@NikhilA8606, what’s the status of your PR? I need to make changes to FacilitySelect, but it looks like that’s going to be removed in your PR. If that’s the case, I can skip the changes. Let me know!

NikhilA8606 commented:
> @NikhilA8606, what’s the status of your PR? I need to make changes to FacilitySelect, but it looks like that’s going to be removed in your PR. If that’s the case, I can skip the changes. Let me know!

Yes, the FacilitySelect component is going to be replaced by the Autocomplete component.

NikhilA8606 committed:
Completed all required changes in ResourceDetailsUpdate
